dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/analytics/dimension/DimensionFilters.java

```diff
@@ -0,0 +1,142 @@
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.analytics.dimension;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.StringTokenizer;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+
+import com.google.common.collect.ImmutableMap;
+
+@Getter
+@NoArgsConstructor( access = AccessLevel.PRIVATE )
+@AllArgsConstructor( access = AccessLevel.PRIVATE )
+public class DimensionFilters implements Predicate<DimensionResponse>
+{
+
+    public static final DimensionFilters EMPTY_DATA_DIMENSION_FILTER = new DimensionFilters()
+    {
+        @Override
+        public boolean test( DimensionResponse dimensionResponse )
+        {
+            return true;
+        }
+    };
+
+    private Collection<SingleFilter> filters;
+
+    public static DimensionFilters of( String filterString )
+    {
+        if ( Objects.isNull( filterString ) || filterString.trim().equals( "" ) )
+        {
+            return EMPTY_DATA_DIMENSION_FILTER;
+        }
+        List<SingleFilter> filters = Arrays.stream( filterString.split( ";" ) )
+            .map( String::trim )
+            .map( SingleFilter::of )
+            .filter( Objects::nonNull )
+            .collect( Collectors.toList() );
+
+        if ( filters.isEmpty() )
+        {
+            return EMPTY_DATA_DIMENSION_FILTER;
+        }
+        return new DimensionFilters( filters );
+    }
+
+    @Override
+    public boolean test( DimensionResponse dimensionResponse )
+    {
+        return filters.stream().allMatch( filter -> filter.test( dimensionResponse ) );
+    }
+
+    @Getter
+    @AllArgsConstructor( access = AccessLevel.PRIVATE )
+    private static class SingleFilter implements Predicate<DimensionResponse>
+    {
+        private static final Map<String, Function<DimensionResponse, ?>> FIELD_EXTRACTORS = ImmutableMap.of(
+            "name", DimensionResponse::getName,
+            "dimensionType", DimensionResponse::getDimensionType,
+            "displayName", DimensionResponse::getDisplayName,
+            "displayShortName", DimensionResponse::getDisplayShortName );
+
+        private static final Map<String, BiFunction<String, String, Boolean>> OPERATOR_MAP = ImmutableMap.of(
+            "eq", String::equalsIgnoreCase,
+            "like", String::contains,
+            "ilike", ( fieldValue, value ) -> fieldValue.toLowerCase().contains( value.toLowerCase() ) );
+
+        private String field;
+
+        private String operator;
+
+        private String value;
+
+        private static SingleFilter of( String filter )
```

Comment:
exactly the same method is used in DimensionalItemFilters class. Maybe you can create small utitlity class to extract it.

Reply:
uff something gone wrong while merging previous PR into this, deleted files has not been deleted. DimensionalItemFilter should have gone, I'll fix it, thank you for spotting